compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaFunctionTypeSymbol.java

```diff
@@ -60,8 +60,13 @@ public BallerinaFunctionTypeSymbol(CompilerContext context, ModuleID moduleID,
     @Override
     public List<ParameterSymbol> parameters() {
         if (this.requiredParams == null) {
-            SymbolFactory symbolFactory = SymbolFactory.getInstance(this.context);
+            // Becomes null for the function typedesc.
+            if (this.typeSymbol.params == null) {
+                this.requiredParams = Collections.emptyList();
+                return this.requiredParams;
```

Comment:
For the compiler and runtime we decided on not setting an empty list for the `function` type descriptor to differentiate between `function` and `function ()`. Shouldn't we do the same for the semantic API? Should this method return `Optional<List<ParameterSymbol>>`?

Also, for the compiler and the runtime we also have a flag `ANY_FUNCTION` that can be used to check if it is `function`. Would the users of the semantic API be able to figure this out directly given that we are returning an empty list here?

Reply:
Need to check that with the others and see. I think having an empty list should be ok. I don't think there'd be any requirement to distinguish between `function` and full form of the function typedesc. Usually the users are interested in details such as the params, return type, meta data etc. The only place this would be relevant AFAIU is the signature.

Reply:
Shall we create an issue to revisit this since `function` is a bit different from other function types?